internal/monitors/telegraf/common/accumulator/accumulator.go

```diff
@@ -57,9 +76,69 @@ func (ac *Accumulator) AddError(err error) {
 	ac.emit.AddError(err)
 }
 
+// AddMetric - adds a metric and will use the configured
+func (ac *Accumulator) AddMetric(m telegraf.Metric) {
+	m.Accept() // TOOD: mark telegraf tracking metrics correctly in emitter
+	ac.emit.AddMetric(m)
+}
+
+// AddMetrics - is a convenience function for adding multiple metrics to the
+// telegraf accumulator
+func (ac *Accumulator) AddMetrics(ms []telegraf.Metric) {
+	for _, m := range ms {
+		ac.AddMetric(m)
+	}
+}
+
+// WithTracking - returns an accumulator that supports tracking metrics
+func (ac *Accumulator) WithTracking(max int) telegraf.TrackingAccumulator {
+	return &TrackingAccumulator{
+		Accumulator: ac,
+		done:        make(chan telegraf.DeliveryInfo, max),
+	}
+}
+
 // NewAccumulator returns a pointer to an Accumulator
 func NewAccumulator(e emitter.Emitter) *Accumulator {
 	return &Accumulator{
 		emit: e,
 	}
 }
+
+// TrackingAccumulator will signal after a metric has been processed.  It indicates
+// if the metric was accepted, rejected, or dropped
+type TrackingAccumulator struct {
```

Comment:
What is the point of this?  I don't see it used anywhere.

Reply:
I'm still trying to understand that myself, they added it and so we have to conform to their interface.  It's a mechanism for input plugins to track the status of a metric they've emitted.